### PR TITLE
Acpi cleanups and up_squared celeron fix (via properly using apic_id from acpi)

### PR DESCRIPTION
**Pull request by tbursztyka**
Various patches to clean a bit ACPI, before expanding its usage.

Note that latest patch fixes up_squared board made of Celeron CPU (the Atom version worked perfectly well without it)

So there are slight differences on how up_squared boards behave depending on their CPU flavor. Can anyone test for the Pentium one?

### CODE REVIEW

File: arch/x86/core/acpi.c [hunk 1]
```diff
@@ -5,6 +5,10 @@
 #include <kernel.h>
 #include <arch/x86/acpi.h>
 
+#define RSDP_UNSEARCHED (void *)0xFFFFFFFF
```
Comment (tbursztyka) [inline arch/x86/core/acpi.c:8]: I am not super confident about this trick (what if the pointer is above 4G, the first 32bits could be made of that value perhaps?). Idea was to save one bool is_found; but ... well it's on x86 with quite a good amount of memory so maybe we don't care.

File: include/arch/x86/acpi.h [hunk 2]
```diff
@@ -8,19 +8,17 @@
...
-	uint32_t sig;
-	uint32_t len;
-	uint8_t rev;
-	uint8_t csum;
-	char oemid[6];
+	uint32_t signature;
+	uint32_t length;
+	uint8_t revision;
+	uint8_t chksum;
+	char oem_id[6];
```
Comment (andyross) [inline include/arch/x86/acpi.h:32]: No complaints about longer symbol names obviously.  Let's just say though that I'm amused about the irony here in ACPI code, an environment that baked a 4-digit symbol name limit into its specification.

File: arch/x86/core/acpi.c [hunk 3]
```diff
@@ -34,7 +34,8 @@ static bool check_sum(struct acpi_sdt *t)
 	for (int i = 0; i < t->length; i++) {
 		sum += p[i];
 	}
-	return sum == 0;
+
+	return !!(sum == 0);
```
Comment (andyross) [inline arch/x86/core/acpi.c:20]: This looks like a needless noop to me. "`sum == 0`" has boolean type already, so double-negating it produces an identity function.

File: arch/x86/core/acpi.c (hunk 1 above)
Comment (andyross) [reply arch/x86/core/acpi.c:8]: IIRC the spec requires that ACPI table address be aligned to 16 bytes (or 8?  I think it's 16).  If you need a value that is guaranteed illegal then anything with some low bits set is likely fine.  This looks good to me.

**Review by andyross — APPROVED** [review]
Some nitpicks.  Looks good.

**Review by dcpleung — APPROVED** [review]

File: arch/x86/core/acpi.c (hunk 3 above)
Comment (andrewboie) [reply arch/x86/core/acpi.c:20]: same, I don't understand this change.

File: arch/x86/core/acpi.c (hunk 1 above)
Comment (andrewboie) [reply arch/x86/core/acpi.c:8]: FWIW I doubt we will ever care about being thrifty with memory on a PC-like system. We have to skip most of the first megabyte of RAM anyway. Keep this in mind when considering any memory optimizations that might not be intuitive. Consider just using NULL, we *never* map the NULL virtual address.

side note: eventually all this legacy stuff needs to go. We really just need to get the RSDP out of UEFI. This kind of legacy probing is not guaranteed to work on a UEFI system. I opened some bugs about this, no need to address this PR

File: arch/x86/core/intel64/cpu.c [hunk 4]
```diff
@@ -121,6 +122,16 @@ void arch_start_cpu(int cpu_num, k_thread_stack_t *stack, int sz,
 	uint8_t vector = ((unsigned long) x86_ap_start) >> 12;
 	uint8_t apic_id = x86_cpu_loapics[cpu_num];
 
+	if (IS_ENABLED(CONFIG_ACPI)) {
+		struct acpi_cpu *cpu;
+
+		cpu = z_acpi_get_cpu(cpu_num);
+		if (cpu != NULL) {
+			/* ToDo: check the flags */
```
Comment (andrewboie) [inline arch/x86/core/intel64/cpu.c:130]: what are the implications of not checking them? suggest filing a GH enhancement if this is something we really want to circle back on.

**Review by tyrantlizard77 — APPROVED** [review]
Broadly speaking LGTM just some comments

File: arch/x86/core/acpi.c (hunk 1 above)
Comment (tbursztyka) [reply arch/x86/core/acpi.c:8]: idea here is to lookup for rsdp only once. So NULL is not the right value to determine if we already looked for it. I'll just have a boolean.

File: arch/x86/core/intel64/cpu.c (hunk 4 above)
Comment (tbursztyka) [reply arch/x86/core/intel64/cpu.c:130]: Actually I fixed this by checking the flags in z_acpi_get_cpu(). In my up², I end up with 4 CPUs in the MADT, with 2 disabled.

Comment (tbursztyka) [reply arch/x86/core/intel64/cpu.c]: Btw, the documentation is not totally clear about the hierarchy of CPU (BSP vs APs). It says ""The MP initialization protocol defines two classes of processors: the bootstrap processor (BSP) and the application processors (APs). Following a power-up or RESET of an MP system, system hardware dynamically selects one of the processors on the system bus as the BSP. The remaining processors are designated as APs."

Does that mean the BSP could be any of the 2 cores then? So not specifically the first CPU in the ACPI MADT?
I quickly checked on my up², getting the bsp apic_id via cpuid instruction. Ok it matches the order we hard-coded in zephyr, but could it be that such order may not be this one? (Or is are the apic_id set after the BSP/APs mitigation process was done... could not find a clear answer on that one).

File: arch/x86/core/acpi.c [hunk 5]
```diff
@@ -31,7 +31,7 @@ static bool check_sum(struct acpi_sdt *t)
 {
 	uint8_t sum = 0, *p = (uint8_t *)t;
```
Comment (nashif) [inline arch/x86/core/acpi.c:13]: t? :-)

**Review by nashif — APPROVED** [review]

File: arch/x86/core/acpi.c (hunk 5 above)
Comment (tbursztyka) [reply arch/x86/core/acpi.c:13]: indeed, some variable could be renamed too

**Pull request merged**